litert/vendors/qualcomm/qnn_manager.h

```diff
@@ -203,12 +207,25 @@ class QnnManager {
 class QnnManager::ContextHandle {
  public:
   ContextHandle(Qnn_ContextHandle_t context_handle, Qnn_ProfileHandle_t profile,
-                QnnContext_FreeFn_t free_fn)
-      : context_handle_(context_handle), profile_(profile), free_fn_(free_fn) {}
+                QnnContext_FreeFn_t free_fn,
+                QnnProfile_FreeFn_t profile_free_fn)
+      : context_handle_(context_handle),
+        profile_(profile),
+        free_fn_(free_fn),
+        profile_free_fn_(profile_free_fn) {}
 
   ~ContextHandle() {
+    if (profile_ && profile_free_fn_) {
+      if (auto status = profile_free_fn_(profile_); status != QNN_SUCCESS) {
+        LITERT_LOG(LITERT_ERROR, "%s", "Failed to free profile handle\n");
+      }
+      profile_ = nullptr;
+    }
     if (context_handle_ && free_fn_) {
-      free_fn_(context_handle_, profile_);
+      if (auto status = free_fn_(context_handle_, profile_);
+          status != QNN_SUCCESS) {
+        LITERT_LOG(LITERT_ERROR, "%s", "Failed to free context handle\n");
+      }
```

Comment:
```suggestion
      }
      context_handle_ = nullptr;
```
You seem to be setting the `profile_` to `nullptr` above, so I'm assuming you forgot to do it here.